source/libs/api.ts

```diff
@@ -85,9 +88,15 @@ export const v3 = mem(async (
 	query: string,
 	options: GHRestApiOptions = v3defaults
 ): Promise<RestResponse> => {
-	const {ignoreHTTPStatus, method, body, headers} = {...v3defaults, ...options};
+	const {ignoreHTTPStatus, method, body, headers, json} = {...v3defaults, ...options};
 	const {personalToken} = await settings;
 
+	// Handle full URLs we might have received from the API itself
+	if (query.startsWith('https://')) {
+		query = new URL(query).pathname.slice(1).replace('api/v3/', '');
+	}
+
+	console.log('Will fetch', api3 + query); // TODO: remove temporary logging
 	const response = await fetch(api3 + query, {
```

Comment:
Just realized I made an error in my code before, this will strip the `search` from the url 🙈 
So it should have been:
```js
const { pathname, search } = new URL(url);
return pathname.slice(1).replace('api/v3/', '') + search;
```

But now that this is in the lib it might be easier to just do:
```js
const url = new URL(query, api3);
await fetch(url)
```

Reply:
I like your second solution! However it needs to verify that query doesn’t start with `/` because it would skip `api/v3` in that case.

Reply:
Added the check :+1: 
*Wasn't sure what the message should be though :smile:*